src/core_functions/scalar/map/map_entries.cpp

```diff
@@ -12,18 +12,37 @@ namespace duckdb {
 static void MapEntriesFunction(DataChunk &args, ExpressionState &state, Vector &result) {
 	auto count = args.size();
 
-	MapUtil::ReinterpretMap(result, args.data[0], count);
+	auto &map = args.data[0];
+	if (map.GetType().id() == LogicalTypeId::SQLNULL) {
+		// Input is a constant NULL
+		auto &validity = FlatVector::Validity(result);
+		validity.SetInvalid(0);
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
```

Comment:
Probably we won't need this anymore either after we have `DEFAULT_NULL_HANDLING`